### PR TITLE
riscv: telink: merge the jump logic.

### PR DESCRIPTION
**Pull request by haiwentelink**
- add the jump logic for dual mode .
- add the support for the b92 and tl3218x.
- add the chip id proc .

### CODE REVIEW

File: boot/zephyr/main.c [hunk 1]
```diff
@@ -319,6 +322,51 @@ static void do_boot(struct boot_rsp *rsp)
...
+
+#define USER_MATTER_PAIR_VAL    0x55  // jump to matter
+
+#define USER_INIT_VAL           0xff  // init state or others will go into zb 
+#define USER_ZB_SW_VAL          0xaa  // jump to matter,use XIP
+#define USER_MATTER_BACK_ZB     0xa0  // only commisiion fail will back to zb 
+
+#define ZB_FW_FLAG_OFFSET       0x20 //telink fw valid flag offset .
+
+const struct device * flash_para_dev = USER_PARTITION_DEVICE;
```
Comment (liucoldstar) [inline boot/zephyr/main.c:350]: 下面的代码可以做些局部的调优。看这个PR
https://github.com/telink-semi/mcuboot/pull/2

File: boot/zephyr/prj.conf [hunk 2]
```diff
@@ -34,3 +34,5 @@ CONFIG_LOG_MODE_MINIMAL=y # former CONFIG_MODE_MINIMAL
 CONFIG_LOG_DEFAULT_LEVEL=0
 ### Decrease footprint by ~4 KB in comparison to CBPRINTF_COMPLETE=y
 CONFIG_CBPRINTF_NANO=y
+### Disabling the banner is to allow cmd tool to better parse the chip ID, avoid interface.
```
Comment (liucoldstar) [inline boot/zephyr/prj.conf]: 建议改成这个内容：
Disable the boot banner to ensure accurate parsing of the Chip ID

File: boot/zephyr/main.c [hunk 3]
```diff
@@ -527,6 +668,8 @@ void main(void)
 
     ZEPHYR_BOOT_LOG_START();
 
+    telink_b9x_mcu_boot_startup();
```
Comment (liucoldstar) [inline boot/zephyr/main.c]: 改成telink_mcu_boot_startup因为支持了tl系列芯片

File: boot/zephyr/main.c (hunk 1 above)
Comment (haiwentelink) [reply boot/zephyr/main.c:350]: already merged.

File: boot/zephyr/prj.conf (hunk 2 above)
Comment (haiwentelink) [reply boot/zephyr/prj.conf]: already changed .

File: boot/zephyr/main.c (hunk 3 above)
Comment (haiwentelink) [reply boot/zephyr/main.c]: merge PR already changed .

**Review by damien0x0023 — APPROVED** [review]

**Pull request merged**